teps/0060-remote-resource-resolution.md

```diff
@@ -239,13 +242,94 @@ users can leverage those shared pipelines.
 
 ## Proposal
 
-<!--
-This is where we get down to the specifics of what the proposal actually is.
-This should have enough detail that reviewers can understand exactly what
-you're proposing, but should not include things like API designs or
-implementation.  The "Design Details" section below is for the real
-nitty-gritty.
--->
+Prior to moving this TEP to `implementable` there will be a considerable amount
+of experimentation required to guage suitability of different approaches. There
+are a lot of open questions wrt syntax, approach and architecture that cannot
+be easily answered without trying ideas with running code.
+
+There are several additional changes that would aid experimentation in the
+pre-`implementable` phase of this TEP:
+
+### Proposal 1: Create an Experimental Resolution Project
+
+This proposal proposes adding a new controller in the [Experimental
+Repo](https://github.com/tektoncd/experimental/) that implements some of
+the ideas described in the [Alternatives section](#alternatives) of this TEP.
+
+This will need a new directory added with a controller & OWNERS configuration.
+If we want to try multiple alternatives (e.g. explore both interceptor-style
+and reconciler-style approaches) then we can either have separate subdirectories
+or branches for each implementation.
+
+### Proposal 2: Disable Tekton Pipelines' "In-Tree" Resolution Logic With a Deployment Arg
+
+A problem we face with introducing an experimental resolver controller is that
+any in-cluster process watching for `TaskRun` and `PipelineRun` resources will
+race with Tekton Pipelines to process them as they're submitted to the kube
+API.
+
+In order to support an experimental controller independently performing
+resolution we could add an arg to Tekton Pipelines' controller (e.g.
+`--experimental-disable-in-tree-resolution`) that completely disables the
+built-in resolution logic.
```

Comment:
i wonder if there's an alternative where we could leverage the "pending" state - e.g. if you want to use the experimental controller, you create "pending" pipelineruns, then the experimental resolution controller can do it's thing (embedding the spec) and remove the "pending" status

Reply:
Yeah agree this is an option for PipelineRuns and I considered it.  We don't have that Pending feature for TaskRuns though.  Should we gate this work on that being decided on and delivered first?

Reply:
ah right - yeah that's kind of a pain 🤔 

at the least i think the _deciding_ part of adding "pending" for taskruns would be very brief in that the PR that was accepted for the pending tep actually was called "TEP-0015 - Add a pending setting to Tekton PipelineRun and TaskRuns" (https://github.com/tektoncd/community/pull/203) (more discussion in the PR that adds the feature https://github.com/tektoncd/pipeline/pull/3522#issuecomment-729886793) - so since we accepted adding pending PipelineRuns, im not sure why we wouldnt accept pending TaskRuns

I agree with you tho, making this TEP depend on that happening would be putting yet another obstacle in the path of this TEP and I'd rather accept the lesser evil of an extra config option than make it take even longer to start experimenting with this

(and as a side note, ive had this awful/wonderful idea recently that we'd be better off making the pending setting more like a semaphore - i.e. what if you have multiple controllers that need to sign off before executing a pending run? maybe there's a better way than that, e.g. a list of explicit controllers you want to process a run before it executes? LONG STORY SHORT i think you're making the right call)

Reply:
> (and as a side note, ive had this awful/wonderful idea recently that we'd be better off making the pending setting more like a semaphore - i.e. what if you have multiple controllers that need to sign off before executing a pending run? maybe there's a better way than that, e.g. a list of explicit controllers you want to process a run before it executes? LONG STORY SHORT i think you're making the right call)

YES.  This is an excellent point.  Very tricky edge case if a user wants both resolution _and_ has their own pending processing logic.

Reply:
(I've also updated the doc to include mention of this alternative approach)